File: Tests/InkFlowKitTests/ViewModels/MockDocumentRepository.swift
import Foundation
@testable import InkFlowKit

@MainActor
final class MockDocumentRepository: DocumentRepository {
    var documents: [InkDocument] = []
    var saveCallCount = 0
    var removeCallCount = 0
    var lastRemovedId: UUID?
    var shouldThrowOnSave = false
    var shouldThrowOnGetAll = false
    var shouldThrowOnGetById = false
    var shouldThrowOnRemove = false

    func getAll() throws -> [InkDocument] {
        if shouldThrowOnGetAll {
            throw MockError.intentional
        }
        return documents.sorted { $0.updatedAt > $1.updatedAt }
    }

    func getById(_ id: UUID) throws -> InkDocument? {
        if shouldThrowOnGetById {
            throw MockError.intentional
        }
        return documents.first { $0.id == id }
    }

    func save(_ document: InkDocument) throws {
        if shouldThrowOnSave {
            throw MockError.intentional
        }
        saveCallCount += 1
        if let index = documents.firstIndex(where: { $0.id == document.id }) {
            documents[index] = document
        } else {
            documents.append(document)
        }
    }

    func remove(_ id: UUID) throws {
        if shouldThrowOnRemove {
            throw MockError.intentional
        }
        removeCallCount += 1
        lastRemovedId = id
        documents.removeAll { $0.id == id }
    }

    enum MockError: Error {
        case intentional
    }
}

File: src/data/MockDocumentRepository.ts
import type { InkDocument } from '../models/InkDocument'
import type { DocumentRepository } from './DocumentRepository'

export class MockDocumentRepository implements DocumentRepository {
  documents: InkDocument[] = []
  saveCallCount = 0
  removeCallCount = 0
  lastRemovedId: string | null = null
  shouldThrowOnSave = false
  shouldThrowOnGetAll = false
  shouldThrowOnGetById = false
  shouldThrowOnRemove = false

  async getAll(): Promise<InkDocument[]> {
    if (this.shouldThrowOnGetAll) throw new Error('getAll failed')
    return [...this.documents].sort((a, b) => b.updatedAt - a.updatedAt)
  }

  async getById(id: string): Promise<InkDocument | null> {
    if (this.shouldThrowOnGetById) throw new Error('getById failed')
    return this.documents.find(d => d.id === id) ?? null
  }

  async save(document: InkDocument): Promise<void> {
    if (this.shouldThrowOnSave) throw new Error('save failed')
    this.saveCallCount++
    const index = this.documents.findIndex(d => d.id === document.id)
    if (index >= 0) {
      this.documents[index] = document
    } else {
      this.documents.push(document)
    }
  }

  async remove(id: string): Promise<void> {
    if (this.shouldThrowOnRemove) throw new Error('remove failed')
    this.removeCallCount++
    this.lastRemovedId = id
    this.documents = this.documents.filter(d => d.id !== id)
  }
}
